schema/pgsql/upgrades/032.sql

```diff
@@ -0,0 +1,61 @@
+ALTER TABLE rule_escalation RENAME CONSTRAINT rule_escalation_rule_id_fkey TO fk_rule_escalation_rule;
+ALTER TABLE rule_escalation RENAME CONSTRAINT rule_escalation_fallback_for_fkey TO fk_rule_escalation_rule_escalation;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_check TO ck_rule_escalation_recipient_one_recipient;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_rule_escalation_id_fkey TO fk_rule_escalation_recipient_rule_escalation;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_contact_id_fkey TO fk_rule_escalation_recipient_contact;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_contactgroup_id_fkey TO fk_rule_escalation_recipient_contactgroup;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_schedule_id_fkey TO fk_rule_escalation_recipient_schedule;
+ALTER TABLE rule_escalation_recipient RENAME CONSTRAINT rule_escalation_recipient_channel_id_fkey TO fk_rule_escalation_recipient_channel;
+ALTER TABLE incident RENAME CONSTRAINT incident_object_id_fkey TO fk_incident_object;
+ALTER TABLE incident_event RENAME CONSTRAINT incident_event_incident_id_fkey TO fk_incident_event_incident;
+ALTER TABLE incident_event RENAME CONSTRAINT incident_event_event_id_fkey TO fk_incident_event_event;
+ALTER TABLE incident_contact RENAME CONSTRAINT key_incident_contact_contact TO uk_incident_contact_contact;
+ALTER TABLE incident_contact RENAME CONSTRAINT key_incident_contact_contactgroup TO uk_incident_contact_contactgroup;
+ALTER TABLE incident_contact RENAME CONSTRAINT key_incident_contact_schedule TO uk_incident_contact_schedule;
+ALTER TABLE incident_contact RENAME CONSTRAINT nonnulls_incident_recipients_check TO ck_incident_contact_one_recipient;
+ALTER TABLE incident_contact RENAME CONSTRAINT incident_contact_incident_id_fkey TO fk_incident_contact_incident;
+ALTER TABLE incident_contact RENAME CONSTRAINT incident_contact_contact_id_fkey TO fk_incident_contact_contact;
+ALTER TABLE incident_contact RENAME CONSTRAINT incident_contact_contactgroup_id_fkey TO fk_incident_contact_contactgroup;
+ALTER TABLE incident_contact RENAME CONSTRAINT incident_contact_schedule_id_fkey TO fk_incident_contact_schedule;
+ALTER TABLE incident_rule RENAME CONSTRAINT incident_rule_incident_id_fkey TO fk_incident_rule_incident;
+ALTER TABLE incident_rule RENAME CONSTRAINT incident_rule_rule_id_fkey TO fk_incident_rule_rule;
+ALTER TABLE incident_rule_escalation_state RENAME CONSTRAINT incident_rule_escalation_state_incident_id_fkey TO fk_incident_rule_escalation_state_incident;
+ALTER TABLE incident_rule_escalation_state RENAME CONSTRAINT incident_rule_escalation_state_rule_escalation_id_fkey TO fk_incident_rule_escalation_state_rule_escalation;
+ALTER TABLE incident_history RENAME CONSTRAINT incident_history_incident_id_fkey1 TO fk_incident_history_incident_rule_escalation_state;
```

Comment:
Interesting, which PostgreSQL version did you use for testing? For me (with PostgreSQL 16.3), the upgrade fails here due to the automatically created constraint being named `incident_history_incident_id_rule_escalation_id_fkey` instead of `incident_history_incident_id_fkey1`. Have we accidentally found a first example of the auto-generated names changing between versions?

Should be easy to work around by creating the new constraint and removing both old names with an additional `IF EXISTS`.

Reply:
God bless my testing setup with PostgreSQL 9.6 and MySQL 5.5 (as well as Boost 1.66 and OpenSSL 1.0.2k). But, jokes aside: I hope this accident pointed out how serious this issue always had been.

## TODO

* [ ] resolve conflicts (@Al2Klimov, shouldn't take very long)